mols2grid/molgrid.py

```diff
@@ -1073,8 +1075,27 @@ def display(
         -------
         view : IPython.core.display.HTML
         """
+        if is_running_within_marimo():
+            use_iframe = True
+
         use_iframe = is_jupyter or use_iframe
         doc = self.render(**kwargs, use_iframe=use_iframe)
+
+        if is_running_within_marimo():
+            import marimo as mo
+
+            if use_iframe:
+                # Render HTML in iframe.
+                iframe = env.get_template("html/iframe.html").render(
+                    width=iframe_width,
+                    height=iframe_height,
+                    allow=iframe_allow,
+                    sandbox=iframe_sandbox,
+                    doc=escape(doc),
+                )
+                return mo.vstack([self.widget, mo.Html(iframe)])
+            return mo.vstack([self.widget, mo.Html(doc)])
```

Comment:
this shouldn't repeat the `use_iframe` block below: nest the `if is_running_within_marimo` right after the `iframe` creation, and either return the `mo.vstack` else the `HTML` object